pinot-core/src/main/java/org/apache/pinot/core/segment/virtualcolumn/VirtualColumnProviderFactory.java

```diff
@@ -35,19 +37,20 @@ public static VirtualColumnProvider buildProvider(String virtualColumnProvider)
     }
   }
 
-  public static void addBuiltInVirtualColumnsToSchema(Schema schema) {
-    if (!schema.hasColumn("$docId")) {
-      schema.addField(new DimensionFieldSpec("$docId", FieldSpec.DataType.INT, true, DocIdVirtualColumnProvider.class));
+  public static void addBuiltInVirtualColumnsToSegmentSchema(Schema schema, String hostName, String segmentName) {
```

Comment:
remove hostname from the argument, and determine the hostname statically in this class.